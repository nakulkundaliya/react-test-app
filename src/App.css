.App{
  position: relative;
}
.content {  
  display: flex;
  flex-direction: column;
  justify-content: center;
}
.form{
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  height: 600px;
}
.input {
  height: 20px;
  width: 300px;
  padding: 10px;
  margin: 5px;
  font-size: 16px;
}
.submit{
  height: 45px;
  width: 100px;
  margin: 7px;
  background-color: aquamarine;
}