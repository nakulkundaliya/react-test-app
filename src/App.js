import React from 'react';
import {
  Route,
  BrowserRouter,
  Switch
} from 'react-router-dom';
import Home from './container/home';
import Projects from './container/projects';
import ProjectDetails from './container/project-details'
import NotFound from './container/not-found'
const App = () => {
  return (
    <BrowserRouter>
      <div>
        <Switch>
          <Route exact path="/" component={Home} />
          <Route exact path="/:name" component={Projects} />
          <Route exact path="/:name/:projectid" component={ProjectDetails} />
          <Route exact component={NotFound} />
        </Switch>
      </div>
    </BrowserRouter>
  );
}

export default App;
