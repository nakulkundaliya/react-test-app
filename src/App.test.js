import React from 'react';
import { shallow, mount } from 'enzyme';

import { createMemoryHistory } from "history";
import { Router } from "react-router";
import App from './App';
import Home from '../src/container/home';
import NotFound from '../src/container/not-found'
let wrapped = shallow(<App />);
describe('render App ', () => {
  it('should render the App correctly', () => {
    expect(wrapped).toMatchSnapshot();
  });

  it('valid path should not redirect to 404', () => {
    const history = createMemoryHistory();

    const wrapper = mount(
      <Router history={history}>
        <App />
      </Router>,
    );
    expect(history.location.pathname).toBe("/");

  });

});