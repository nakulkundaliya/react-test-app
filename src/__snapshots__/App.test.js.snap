// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render App  should render the App correctly 1`] = `ShallowWrapper {}`;
